libbindgen/src/ir/context.rs

```diff
@@ -1215,15 +1219,18 @@ impl<'ctx, 'gen> Iterator for AssertNoDanglingItemIter<'ctx, 'gen>
             None => {
                 // We've traversed everything reachable from the previous
                 // root(s), see if we have any more roots.
+                //
+                // FIXME(emilio): Yay for exponential algorithms?
```

Comment:
As discussed on irc, this commit needs some rework.

* The real issue is that we were inserting the wrong item into `to_iterate`, because we are tacking paths from parent -> child in `seen` (which doesn't make sense since a parent has many children, but a child can only have one parent) rather than child -> parent, right?
* We should be able to continue using the paths as a seen set, once we fix the issue where keys and values were backwards
* This is not exponential, it is quadratic. Once this commit is reworked / recommented, would you mind filing a good-first-bug to fix this so that it is not quadratic anymore?